internal/app/machined/pkg/controllers/block/mount.go

```diff
@@ -232,6 +234,15 @@ func (ctrl *MountController) Run(ctx context.Context, r controller.Runtime, logg
 						mountStatus.TypedSpec().EncryptionProvider = volumeStatus.TypedSpec().EncryptionProvider
 						mountStatus.TypedSpec().ReadOnly = mountRequest.TypedSpec().ReadOnly
 						mountStatus.TypedSpec().ProjectQuotaSupport = volumeStatus.TypedSpec().MountSpec.ProjectQuotaSupport
+						mountStatus.TypedSpec().Detached = mountRequest.TypedSpec().Detached
+
+						// This needs to be set through accessor, and is not guaranteed to resolve to a valid root.
+						mount, ok := ctrl.activeMounts[mountRequest.Metadata().ID()]
+						if ok && mount.point != nil {
+							mountStatus.TypedSpec().SetRoot(mount.point.Root())
+						} else {
+							mountStatus.TypedSpec().SetRoot(&xfs.OSRoot{Shadow: filepath.Join(rootPath, mountTarget)})
```

Comment:
quick question - why can't we use `mount.point.Root()` here always?

Reply:
ok, got you... probably we should do this refactoring to make all of them have this (doesn't have to be this PR of course0